### ISSUE TITLE
Returning gradients for yaw optimization

### ISSUE THREAD

**Issue by petebachant** [NREL/floris#219]
I see that this feature is in the roadmap. Has there been any discussion on how to implement, or maybe a WIP branch?

**Comment by bayc** [comment]
The discussion so far as centered around using Jax (specifically autograd within Jax), and we are currently performing a refactoring of FLORIS in which we are working to enable all the models to be Jax friendly. As for a timeline, we are hoping to have the refactoring done sometime in August, after which the gradients would shortly (hopefully) be available.

**Comment by petebachant** [comment]
> Do you have any experience with returning gradients from more complex code such as FLORIS?

I don't. Just now starting to explore available tools like `algopy`. I also see that @andrewning's group at BYU had modified some version of FLORIS to return gradients in a 2017 paper. Is that code floating around anywhere?

**Comment by bayc** [comment]
I don't have access to it directly, but can reach out and let you know what I find.

**Comment by jaredthomas68** [comment]
@petebachant I'm in the FLOWLab (@andrewning research group). The code you are referring to is still in existence, but is not currently maintained and is in some disrepair if you are still interested in it you can access it at https://github.com/byuflowlab/PlantEnergy and https://github.com/byuflowlab/gaussian-wake. 

We are currently developing a julia version of floris called FLOWFarm, where the gradients can be easily obtained. You can take take a look at https://github.com/byuflowlab/FLOWFarm.jl. We have not used this code for yaw optimization very much, but it does work.

**Comment by bayc** [comment]
@petebachant Following up to see if you made any progress using `algopy` or any other solution on this front. I have tried implementing Jax, but haven't had the results I have hoped for yet.

**Comment by petebachant** [comment]
@bayc nope, unfortunately I haven't gotten around to it. It sounds like the serial refine optimization in 3.0 may negate the need for computing gradients for optimization. Do you guys have any benchmarking on differences in AEP gain between serial refine and gradient based optimization?
